lesson02.md

```diff
@@ -0,0 +1,495 @@
+<!-- .slide: id="lesson2" -->
+
+# Basic Frontend - Fall 2021
+
+Lesson 2, Thursday, 2021-09-16
+
+---
+
+### Recap
+
+What did we talk about last lesson?
+
+---
+
+### Recap: data types
+
+Number:
+```js
+42
+-3.1415
+```
+
+String:
+```js
+"John"
+'Berlin'
+```
+
+Boolean:
+```js
+true
+false
+```
+
+---
+
+### Recap operators
+
+| Group | Operators | Example |
+| ----- | --------- | ------- |
+| Numerical Operators | `+ - * / **` | `5 + 4 * 3`<br/>`7 / 2 - 2`<br/>`"Hello" + " World"` |
+| Comparison Operators | `=== !== < > <= >=` | `30 !== 25`<br/>`20 >= 18` |
+
+---
+
+### Try it out
+
+Now, try to enter some data to the JavaScript console, but think of the data type first:
+
+* Your name
+* Your shoe size
+* Wether you think it's ok to put pineapple on pizza
+
+---
+
+### Try it out: Solution
+
+```js
+"Carlo"
+45
+false
+```
+
+---
+
+### Quiz time
+
+The `typeof` operator in JavaScript outputs the data type of a value. Can you guess the output?
+
+```js
+typeof 42;
+```
+```js
+"number"
+```
+<!-- .element: class="fragment" -->
+
+```js
+typeof "hello";
+```
+```js
+"string"
+```
+<!-- .element: class="fragment" -->
+
+```js
+typeof true;
+```
+```js
+"boolean"
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### Quiz time
+
+```js
+typeof (1 + 1);
```

Comment:
above we use `typeof` without parentheses. Maybe it's better to be consistent and use it always without?

Reply:
I think it's a bit tricky to remove the parentheses for this case because of the addition. If it's `typeof 1 + 1;`, it will be 'number1'. Variables come later in the slides so we can't save it in a variable first as well.